tests/ftests/084-sudo-cgcreate_systemd_scope_pid.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# Test to create a systemd scope with pid using cgcreate
+#
+# Copyright (c) 2023 Oracle and/or its affiliates.
+# Author: Kamalesh Babulal <kamalesh.babulal@oracle.com>
+#
+
+from cgroup import Cgroup
+from process import Process
+from libcgroup import Mode
+from run import RunError
+from log import Log
+import consts
+import ftests
+import sys
+import os
+
+CONTROLLERS = ['cpu', 'pids']
+SLICE = 'libcgroup.slice'
+CGNAME1 = os.path.join(SLICE, '084cgcreate1.scope')
+CGNAME2 = os.path.join(SLICE, '084cgcreate2.scope')
+
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if config.args.container:
+        result = consts.TEST_SKIPPED
+        cause = 'This test cannot be run within a container'
+        return result, cause
+
+    if Cgroup.get_cgroup_mode(config) != Mode.CGROUP_MODE_UNIFIED:
+        result = consts.TEST_SKIPPED
+        cause = 'This test requires the unified cgroup hierarchy'
+
+    return result, cause
+
+
+def setup(config):
+    pass
+
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    #
+    # Test 1: Pass invalid task pid as scope_pid and systemd scope creation
+    #         should fail
+    #
+    try:
+        Cgroup.create_and_validate(config, CONTROLLERS, CGNAME1, create_scope=True,
+                                   scope_pid=1000000)
+    except RunError as re:
+        if 'Process with ID 1000000 does not exist' not in str(re):
+            raise re
+
+    #
+    # Test 2: Pass a valid task pid as scope_pid and systemd scope creation
+    #         will succeed. Read the scope cgroup.procs to find if the task
+    #         passed by us was used as scope task.
+    #
+    scope_pid = config.process.create_process(config)
+    Cgroup.create_and_validate(config, CONTROLLERS, CGNAME1, create_scope=True, scope_pid=scope_pid)
+
+    try:
+        pid = int(Cgroup.get(config, None, CGNAME1, setting='cgroup.procs',
+                             print_headers=False, values_only=True, ignore_systemd=True))
+        # use the pid variable so that lint is happy
+        Log.log_debug('Cgroup {} has pid {}'.format(CGNAME1, pid))
+    except RunError:
+        result = consts.TEST_FAILED
+        cause = 'Failed to read pid in {}\'s cgroup.procs'.format(CGNAME1)
+        return result, cause
+
+    if scope_pid != pid:
```

Comment:
I'm not sure how Python isn't yelling at us here.  `pid` is declared inside of the `try` statement above, but is being tested outside of it.  If you move this `if` inside of the `try` statement, I think you can remove the `log_debug` message above.